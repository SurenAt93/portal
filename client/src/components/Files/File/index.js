import React from 'react';

import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

// Icons
import FileIcon from '@material-ui/icons/Notes';

const File = ({ name }) => (
  <ListItem className="files__item">
    <div className="files__summary">
      <ListItemIcon><FileIcon /></ListItemIcon>
      <ListItemText primary={name}/>
    </div>
  </ListItem>
);

export default File;
