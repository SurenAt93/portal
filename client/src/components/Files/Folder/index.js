import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Collapse from '@material-ui/core/Collapse';

// Icons
import FolderIcon from '@material-ui/icons/Folder';
import FolderIconOpen from '@material-ui/icons/FolderOpen';

class Folder extends Component {
  state = { open: false };

  handleClick = _ => this.setState(state => ({ open: !state.open }));

  render() {
    const { name, path, children, drow } = this.props;

    return (
      <Fragment>
        <ListItem
          button
          onContextMenu={this.props.handleContextMenuOpen}
          onClick={this.handleClick} className="files__item"
        >
          <div className="files__summary">
            <ListItemIcon>
              {
                this.state.open
                  ? <FolderIconOpen />
                  : <FolderIcon />
              }
            </ListItemIcon>
            <ListItemText primary={name}/>
          </div>
        </ListItem>
        <Collapse
          className="files__folder--collapse"
          in={this.state.open}
          timeout="auto"
          unmountOnExit
        >
          <List className="files__folder" dense={true}>
            {children.map(child => drow(child, path))}
          </List>
        </Collapse>
      </Fragment>
    );
  }
}

Folder.propTypes = {
  name: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  children: PropTypes.array.isRequired,
  drow: PropTypes.func.isRequired,
  handleContextMenuOpen: PropTypes.func.isRequired,
};

export default Folder;

