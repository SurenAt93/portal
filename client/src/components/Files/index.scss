.files {
  width: 100%;
  height: 100%;
  background-color: #202124;
  overflow: scroll;
  white-space: nowrap;

  ul {
    height: 100%;
    width: fit-content;
  }

  &__item {
    flex-direction: column;
    align-items: flex-start !important;

    * {
      margin: 0;
      padding: 0 !important;
    }
  }

  &__folder {
    margin-left: 15px !important;
  }

  &__summary {
    display: flex;
    align-items: center;
    > svg {
      margin-right: 10px;
    }
  }
}