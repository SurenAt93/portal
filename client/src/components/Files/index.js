import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import List from '@material-ui/core/List';

// Components
import File from './File';
import Folder from './Folder';

// Styles
import './index.scss';

class Files extends PureComponent {
  drow = (
    { name, children },
    parent,
    path = parent ? `${parent}${name}${children ? '/' : ''}` : name,
  ) => children
    ? <Folder
      key={path}
      path={path}
      name={name}
      children={[...children].sort((a, b) => +!a.children - +!b.children)}
      drow={this.drow}
      handleContextMenuOpen={this.props.handleFolderContextMenuOpen}
    />
    : <File
      handleFileOpen={this.props.handleFileOpen}
      key={path}
      path={path}
      name={name}
      open={this.props.openFilePath === path}
      handleContextMenuOpen={this.props.handleFileContextMenuOpen}
    />;

  render() {
    const { data } = this.props;

    return (
      <div className="files">
        {
          data
            ? (
              <List dense={true}>
                {this.drow(data)}
              </List>
            )
            : 'Empty...'
        }
      </div>
    );
  }
}

Files.propTypes = {
  data: PropTypes.object,
  openFilePath: PropTypes.string,
  handleFileOpen: PropTypes.func,
  handleFolderContextMenuOpen: PropTypes.func.isRequired,
  handleFileContextMenuOpen: PropTypes.func.isRequired,
};

Files.defaultProps = {
  handleFileOpen: _ => {},
};

export default Files;
