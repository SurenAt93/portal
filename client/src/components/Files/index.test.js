import React from 'react';
import Files from '.';

describe('<Files />', () => {
  it('should render without props', () => {
    shallow(<Files />);
  });

  it('should render \'Empty\' in case of it renders without props', () => {
    const component = shallow(<Files />);

    expect(component).toMatchSnapshot();
  });
});
