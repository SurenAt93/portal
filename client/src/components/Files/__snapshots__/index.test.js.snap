// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Files /> should render 'Empty' in case of it renders without props 1`] = `
<div
  className="files"
>
  Empty...
</div>
`;
